### ISSUE TITLE
Filter empty indicator dates in Reports

### ISSUE THREAD

**Issue by hamza-vd** [OpenSRP/opensrp-client-giz-malawi#462]
In reference to #456, dates that don't have any data should be filtered out. This can be achieved in 2 ways:

1. Loop on the dates fetched and query data for each date and hide the dates which have  0 indicator values.
2. Update reporting library with a new SQL query which will return only those dates which have values.

**Note:**
The first method will have the least LOE here. The second method will take more than a day to implement. The only drawback of method 1 is that the loading time for the daily indicators list might increase.

**Comment by AnnieMungai** [comment]
@hamza-vd is this something we can due better in this project phase?

**Comment by hamza-vd** [comment]
@AnnieMungai 
Yes, we can have this enhancement, but if the current implementation is working fine then we can keep it as is. I mean, it should be a low-priority task. 
cc: @dubdabasoduba
